modules/rpc/rpc.js

```diff
@@ -87,26 +97,112 @@ function mkDir(dirPath, root) {
   return !dirs.length || mkDir(dirs.join(path.sep), root);
 }
 
-let auth = [];
+/*
+** handle RPC call response
+*/
+function cb_handleRequestResponse (res, cb) {
+  var data = '';
+  res.setEncoding('utf8');
+  res.on('data', chunk => {data += chunk});
+  res.on('end', () => {
+    if (res.statusCode === 401) {
+      cb(res);
+      return ;
+    }
+    data = JSON.parse(data);
+    cb(null, data);
+  });
+}
+
+/*
+** execute RPC call
+*/
+function rpcCall (method, params, auth, cb) {
+  var postData = JSON.stringify({
+    method: method,
+    params: params,
+    id: '1'
+  });
+
+  var options = {
+    hostname: HOSTNAME,
+    port: PORT,
+    path: '/',
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'Content-Length': postData.length
+    },
+    auth: auth ? auth[0] + ':' + auth[1] : undefined
+  }
+
+  var req = http.request(options, res => cb_handleRequestResponse(res, cb));
+  req.on('error', e => cb(e));
+  req.setTimeout(TIMEOUT, e => {
+    cb(e);
+    return (req.abort());
+  });
+  req.write(postData);
+  req.end();
+}
 
-if (fs.existsSync(COOKIE_FILE))
-  auth = fs.readFileSync(COOKIE_FILE, 'utf8').split(':');
-console.log(COOKIE_FILE);
-//else TODO: No cookie file...
+/*******************************/
+/****** Public functions *******/
+/*******************************/
 
-rpc.init('localhost', TESTNET_PORT, auth[0], auth[1]);
+/*
+** returns the current RPC cookie
+** RPC cookie is regenerated at every particld startup
+*/
+function getCookie(testnet) {
 
-// This is a factory function that returns an Observable
-function createObservable(event, method, params) {
-  return Observable.create(observer => {
-    rpc.call(method, params, (error, response) => {
+  const COOKIE_FILE = findCookiePath() + `${testnet ? '/testnet' : ''}/.cookie`;
+  let auth = [];
+
+  if (fs.existsSync(COOKIE_FILE)) {
+    auth = fs.readFileSync(COOKIE_FILE, 'utf8').split(':');
+  } else {
+    auth = undefined;
+    console.error('could not find cookie file !');
+  }
+  return (auth)
+}
+exports.getCookie = getCookie;
+
+/*
+** prepares `backend-rpccall` to receive RPC calls from the renderer
+*/
+function init(options) {
+
+  HOSTNAME = options.rpcbind;
+  PORT = options.port;
+
+  // This is a factory function that returns an Observable
+  function createObservable(event, method, params) {
+    let auth = getCookie(options.testnet);
```

Comment:
Auth could potentially use -rpcuser, -rpcpassword or -rpcauth.
Just need to bare that in mind..